new-packages/ts-project/src/types.ts

```diff
@@ -0,0 +1,48 @@
+import type { SourceFile } from 'typescript';
+
+export interface ExtractionContext {
+  errors: ExtractionError[];
+  sourceFile: SourceFile;
+}
+
+// TODO: add error location/span
+export type ExtractionError =
+  | {
+      type: 'state_unhandled';
+      node: unknown;
+    }
+  | {
+      type: 'state_property_unhandled';
+      node: unknown;
+    }
+  | {
+      type: 'property_key_no_roundtrip';
+      node: unknown;
+    }
+  | {
+      type: 'property_key_unhandled';
+      propertyKind: 'computed' | 'private';
+      node: unknown;
+    };
+
+export type ExtractorStateConfig = {
```

Comment:
I think that we should aim for compatibility with the `DigraphDef` and not `DigraphData`. To make it happen we might have to make the digraph's initialization more flexible to handle this but there is a chance that we'll be able to handle patch-based diffs way easier. If we keep using `DigraphDef` here then we won't make anything easier - apart from the initialization of the digraph. We also want to "seed" unique ids with the ones generated here but `DigraphData` doesn't even have them (`DigraphDef` has)